src/network/networkd-dhcp-server.c

```diff
@@ -397,6 +399,17 @@ int dhcp4_server_configure(Link *link) {
                         return log_link_error_errno(link, r, "Failed to set DHCPv4 option: %m");
         }
 
+        ORDERED_HASHMAP_FOREACH(static_lease, link->network->dhcp_static_leases_by_section) {
+                /* Static lease with an empty or omitted address is a valid entry,
+                * it removes any static lease with the specified mac address. */
+                if (static_lease->address.s_addr == 0)
+                        r = sd_dhcp_server_remove_static_lease(link->dhcp_server, static_lease->client_id, static_lease->client_id_size);
+                else
```

Comment:
As said before, please drop this removal logic. https://github.com/systemd/systemd/pull/15556#discussion_r562084590 does not mean to implement this logic, but handle an empty string in conf parsers, and the new conf parsers you implemented already handle an empty string gracefully.